PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -1258,16 +1258,25 @@ SavingStatus FileManager::saveBuffer(BufferID id, const TCHAR * filename, bool i
 	}
 }
 
-size_t FileManager::nextUntitledNewNumber() const
+size_t FileManager::nextUntitledNewNumber(bool toMainOrSubEditView) const
 {
 	std::vector<size_t> usedNumbers;
 	for (size_t i = 0; i < _buffers.size(); i++)
 	{
 		Buffer *buf = _buffers.at(i);
 		if (buf->isUntitled())
 		{
+			bool isVisible = false;
+			for (size_t k = 0; k < buf->_referees.size(); k++)
+			{
+				if (buf->_referees[k]->isVisible())
+				{
+					isVisible = true;
+					break;
+				}
+			}
 			// if untitled document is invisible, then don't put its number into array (so its number is available to be used)
-			if ((buf->_referees[0])->isVisible())

```

Comment:
Please use the following code:
```cpp
		if (buf->isUntitled())
		{
			bool isVisible = false;
			for (size_t k = 0; k < buf->_referees.size(); k++)
			{
				if (buf->_referees[k]->isVisible())
				{
					isVisible = true;
					break;
				}
			}

			if (isVisible)
			{
				if (buf->indexOfReference(_pNotepadPlus->_pEditView) > -1 || buf->indexOfReference(_pNotepadPlus->_pNonEditView) > -1)
				{
					generic_string newTitle = ((NppParameters::getInstance()).getNativeLangSpeaker())->getLocalizedStrFromID("tab-untitled-string", UNTITLED_STR);
					TCHAR* numberStr = buf->_fileName + newTitle.length();
					int usedNumber = _wtoi(numberStr);
					usedNumbers.push_back(usedNumber);
				}
			}
		}
```
to replace the old code:
```cpp
		if (buf->isUntitled())
		{
			// if untitled document is invisible, then don't put its number into array (so its number is available to be used)
			if ((buf->_referees[0])->isVisible())
			{
				generic_string newTitle = ((NppParameters::getInstance()).getNativeLangSpeaker())->getLocalizedStrFromID("tab-untitled-string", UNTITLED_STR);
				TCHAR *numberStr = buf->_fileName + newTitle.length();
				int usedNumber = _wtoi(numberStr);
				usedNumbers.push_back(usedNumber);
			}
		}
```

The other modification is not necessary.